opsgenie/resource_opsgenie_incident_template_test.go

```diff
@@ -0,0 +1,167 @@
+package opsgenie
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	ogClient "github.com/opsgenie/opsgenie-go-sdk-v2/client"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/incident"
+)
+
+var randomName = acctest.RandString(6)
+
+func init() {
+	resource.AddTestSweepers("opsgenie_incident_template", &resource.Sweeper{
+		Name: "opsgenie_incident_template",
+		F:    testSweepIncidentTemplate,
+	})
+}
+
+func testSweepIncidentTemplate(region string) error {
+	meta, err := sharedConfigForRegion()
+	if err != nil {
+		return err
+	}
+	client, err := incident.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+	if err != nil {
+		return err
+	}
+	if result != nil {
+		for _, value := range result.IncidentTemplates["incidentTemplates"] {
+			incidentName := "Incident Template Name-" + randomName
+			if strings.HasPrefix(value.Name, incidentName) {
```

Comment:
we can get name or id from template ?

Reply:
Changes done. removed randomName variable.